packages/main/src/plugin/container-registry.ts

```diff
@@ -347,6 +347,27 @@ export class ContainerProviderRegistry {
     return flatttenedImages;
   }
 
+  async pruneImages(engineId: string): Promise<void> {
+    this.telemetryService.track('pruneImages');
+    // We have to use two different API calls for pruning images, because the Podman API does not respect the 'dangling' filter
+    // and instead uses 'all' and 'external'. See: https://github.com/containers/podman/issues/11576
+    // so for Dockerode we'll have to call pruneImages with the 'dangling' filter, and for Podman we'll have to call pruneImages
+
+    // PODMAN:
+    // Have to use podman API directly for pruning images
+    const provider = this.internalProviders.get(engineId);
+    if (provider?.libpodApi) {
+      return this.getMatchingPodmanEngine(engineId).pruneAllImages(true);
```

Comment:
we don't have the method on Dockerode ? I would not add the method using libpod else it means it'll work only with posman e gine

Reply:
We do, but it doesn't work with Podman. See: https://github.com/containers/podman/issues/11576

Podman doesn't respect the filters being passed into `pruneImages`. So we have to make our own call with the podman engine. 

See the specific comment here: https://github.com/containers/podman/issues/11576#issuecomment-936172087

Try it with Dockerode, it doesn't work with pruneImages.

Reply:
the issues that you referenced was about the libpod endpoint

I created one for the compat endpoint: https://github.com/containers/podman/issues/17614